code/modules/client/preferences_toggles.dm

```diff
@@ -214,7 +214,10 @@
 	set desc = "Toggles which special roles you would like to be a candidate for, during events."
 	var/role_flag = be_special_flags[role]
 	if(!role_flag)	return
-	prefs.be_special ^= role_flag
+	if(role_flag in prefs.be_special)

```

Comment:
REQUESTED CHANGE (from Oranges and MSO):
Change this to not use a basic byond list anymore, so that we get a *nicer* UI that actually tells us which roles are enabled and which are disabled.

Then perhaps in a later PR allow preferences to be opened anytime so players get the same editing window+UI as in the lobby, you know, the not shit one.